src/AspNetCoreModuleV2/AspNetCore/Inc/appoffline.h

```diff
@@ -14,92 +16,64 @@ class APP_OFFLINE_HTM
-        HANDLE         handle = INVALID_HANDLE_VALUE;
 
-        handle = CreateFile(m_Path.QueryStr(),
-            GENERIC_READ,
-            FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
-            NULL,
-            OPEN_EXISTING,
-            FILE_ATTRIBUTE_NORMAL,
-            NULL);
+        HandleWrapper<InvalidHandleTraits> handle = CreateFile(m_Path.QueryStr(),
+                                                               GENERIC_READ,
+                                                               FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                                                               NULL,
+                                                               OPEN_EXISTING,
+                                                               FILE_ATTRIBUTE_NORMAL,
+                                                               NULL);
 
         if (handle == INVALID_HANDLE_VALUE)
         {
-            if (HRESULT_FROM_WIN32(GetLastError()) == ERROR_FILE_NOT_FOUND)
+            if (GetLastError() == ERROR_FILE_NOT_FOUND)
             {
-                fResult = FALSE;
+                return false;
             }
 
             // This Load() member function is supposed be called only when the change notification event of file creation or file modification happens.
             // If file is currenlty locked exclusively by other processes, we might get INVALID_HANDLE_VALUE even though the file exists. In that case, we should return TRUE here.
-            goto Finished;
+            LOG_LAST_ERROR_IF(true);
+            return true;
         }
 
-        if (!GetFileSizeEx(handle, &li))
+        if (LOG_LAST_ERROR_IF(!GetFileSizeEx(handle, &li)))
         {
-            goto Finished;
+            return true;
         }
 
         if (li.HighPart != 0)
         {
             // > 4gb file size not supported
-            // todo: log a warning at event log
-            goto Finished;
+            LOG_ERRORF("app_ofline file larger than 4gb");
+            return true;
         }
 
-        DWORD bytesRead = 0;
-
         if (li.LowPart > 0)
         {
-            pszBuff = new CHAR[li.LowPart + 1];
+            DWORD bytesRead = 0;
+            std::unique_ptr<char[]> pszBuff(new CHAR[li.LowPart + 1]);
 
-            if (ReadFile(handle, pszBuff, li.LowPart, &bytesRead, NULL))
+            if (LOG_LAST_ERROR_IF(!ReadFile(handle, pszBuff.get(), li.LowPart, &bytesRead, NULL)))
             {
-                m_Contents.Copy(pszBuff, bytesRead);
+                return true;
             }
-        }
 
-    Finished:
-        if (handle != INVALID_HANDLE_VALUE)
-        {
-            CloseHandle(handle);
-            handle = INVALID_HANDLE_VALUE;
+            LOG_IF_FAILED(m_Contents.Copy(pszBuff.get(), bytesRead));
         }
 
-        if (pszBuff != NULL)
-        {
-            delete[] pszBuff;
-            pszBuff = NULL;
-        }
-
-        return fResult;
+        return true;
```

Comment:
Lot of return true here. Didn't see an easy way to reduce it besides nesting ifs.